cloudman/values.yaml

```diff
@@ -62,6 +62,89 @@ helmsman_config:
+      repo: cloudve
+      chart: nextcloud
+      display_name: File Management (NextCloud)
+      context:
+        oidc_client_secret: '{{.Values.oidc_clients.nextcloud.client_secret}}'
+        oidc_client_id: '{{ .Values.oidc_clients.nextcloud.client_id}}'
+        domain: '{{ .Values.global.domain }}'
+      template: |
+        ingress:
+          enabled: true
+          access_path: "/{{context.project.access_path}}/nextcloud"
+          hosts:
+            - name: {{context.domain}}
+              path: /{{context.project.access_path}}/nextcloud
+
+        nextcloud:
+          configs:
+            path.config.php: |
+              <?php
+              $CONFIG = array (
+                'overwriteprotocol' => 'https',
+                'overwritewebroot' => '{{context.project.access_path}}/nextcloud',
+                'overwritehost'     => '{{context.domain}}',
+                'overwrite.cli.url' => 'https://{{context.domain}}/{{context.project.access_path}}/nextcloud',
+                'skeletondirectory' => '',
+                'allow_user_to_change_display_name' => false,
+                'lost_password_link' => 'disabled',
+                'oidc_login_default_quota' => 1000000000,
+                'oidc_login_client_id' => '{{context.oidc_client_id}}',
+                'oidc_login_client_secret' => '{{context.oidc_client_secret}}',
+                'oidc_login_provider_url' => 'https://{{context.domain}}/auth/realms/master',
+                'oidc_login_logout_url' => 'https://{{context.domain}}/auth/realms/master/protocol/openid-connect/logout?redirect_uri=https%3A%2F%2F{{context.domain}}%2F{{context.project.access_path}}%2Fnextcloud%2F',
+                'oidc_login_scope' => 'openid profile',
+                'oidc_login_auto_redirect' => true,
+                'oidc_login_redir_fallback' => true,
+                'oidc_login_use_external_storage' => true,
+                'oidc_login_disable_registration' => false,
+                'oidc_login_attributes' => array(
+                  'id' => 'preferred_username',
+                  'mail' => 'email',
+                ),
+              );
+
+          host: "{{context.domain}}"
+          extraVolumes:
+            - name: project-data
+              persistentVolumeClaim:
+                claimName: "{{context.project.namespace}}-project-data-pvc"
+            - name: public-data
+              persistentVolumeClaim:
+                claimName: "{{context.project.namespace}}-public-data-pvc"
+
+
+          extraVolumeMounts:
+            - name: project-data
+              mountPath: /gvl/projects/current
+            - name: public-data
+              mountPath: /gvl/public
+
+        persistence:
+          enabled: false
+
+        lifecycle:
+          postStartCommand:
+            - su
+            - www-data
+            - -s
+            - /bin/bash
+            - -c
+            - > 
+              PATH=/var/www/html:$PATH;
+              occ app:update --all;
```

Comment:
Does this update OwnCloud? Do you know how often these updates are? My current thinking is that we should allow application updates by swapping out the container itself. That way, we have full reproducibility/reduce chances of breakage until the container is explicitly upgraded to a newer version.

Reply:
This only updates the Nextcloud apps. Nextcloud is left untouched. `occ:upgrade` runs any necessary migrations.